staging/src/k8s.io/apiserver/pkg/registry/generic/rest/streamer.go

```diff
@@ -61,11 +64,28 @@ func (s *LocationStreamer) InputStream(ctx context.Context, apiVersion, acceptHe
 	if transport == nil {
 		transport = http.DefaultTransport
 	}
+	requestAddress := s.Location.Host
+	klog.Infof("Sending request to %q.", requestAddress)
+
+	if httpTransport, ok := transport.(*http.Transport); ok {
+		networkContext := server.NetworkContext{ConnectivityServiceName: "cluster"}
+		contextDialer, err := server.Lookup(networkContext)
+		if err != nil {
+			return nil, false, "", fmt.Errorf("failed to get connection for %s, got %v", s.Location.String(), err)
+		}
+		httpTransport.DialContext = contextDialer
```

Comment:
what if httpTransport.DialContext (or Dial) is already set?

Reply:
Switched to using wrapper